### DISCUSSION TITLE
Adding an additional MED177

### DISCUSSION THREAD

**Discussion by MasterCodeIT** [jglim/CaesarSuite#39] [General]
Have an ECU that is not listed on the unlockecu db. Just got my hands on the dll. Would like to add it for the community. 

Dropped the DLL in IDA, found the GenerateKeyExOpt in exports (screen shot below). I see the security 5 and the security 0B. But not sure where to go from here.

![image](https://user-images.githubusercontent.com/73858519/136465019-11232db7-4502-4472-aad3-d4f1f223d162.png)

**Comment by jglim** [comment]
I've written a short guide a while ago for you in this [thread](https://github.com/jglim/CaesarSuite/discussions/32#discussioncomment-885901). The guide outlines the specific reverse-engineering process for security DLLs. 

I do not have the capacity to teach you how to reverse engineer (and this might not be the best place to pick up the basics). Other folks have written better, general guides on how to work on x86 binaries, such as:

https://puri.sm/posts/primer-to-reverse-engineering/
https://sensepost.com/blogstatic/2014/01/SensePost_crash_course_in_x86_assembly-.pdf

Once you have the foundations in place, I would suggest trying to replicate the process in the previously mentioned [thread](https://github.com/jglim/CaesarSuite/discussions/32#discussioncomment-885901) for `MED1775_MED1775_12_42_00.dll`. In that thread, I have provided examples on what the key parameter looks like for level 9, where you might be able to compare your reverse-engineered results with a reference.
